SCNG-2

mlukac-senacor opened the pull request:
Side navigation and toolbar.

mlukac-senacor committed:
SCNG-2
Side navigation and toolbar.

cgolian commented on src/app/app.module.ts:
```diff
@@ -20,6 +22,11 @@ const appRoutes: Routes = [
   },
+  {
+    path: 'dashboard',
+    component: DashboardComponent,
```
I would prefer Dashboard module to be lazy loaded here.

cgolian commented on src/app/login/components/login-form/login-form.component.ts:
```diff
@@ -1,7 +1,7 @@
-import {Component, EventEmitter, OnInit, Output} from '@angular/core';
-import {FormBuilder} from '@angular/forms';
+import { Component, EventEmitter, OnInit, Output } from '@angular/core';
```
Changes like this one are not really necessary...

cgolian commented on src/app/dashboard/components/user-info/user-info.component.ts:
```diff
@@ -0,0 +1,23 @@
+export class UserInfoComponent implements OnInit {
+
+    photo: string;
+    firstName = "Michaela";
```
Even though we don't have a way to fetch user info right now, firstName and lastName should be `@Input()` s

cgolian commented on src/app/dashboard/components/appbar/appbar.component.ts:
```diff
@@ -0,0 +1,31 @@
+
+    sidenavToggleButtonVisible: boolean;
+
+    constructor(public breakpointObserver: BreakpointObserver) {
```
I am not sure if this has to be public or not, however in TS public is the default access modifier so it does not have to be written explicitely (and we are writing it that way in other components)

cgolian commented on src/app/dashboard/components/user-info/user-info.component.ts:
```diff
@@ -0,0 +1,23 @@
+import { OnInit, Component } from '@angular/core';
+
+@Component({
+    selector: 'userinfo',
```
All of our component selectors use app prefix

cgolian commented on src/app/dashboard/components/nav-item/nav-item.component.ts:
```diff
@@ -0,0 +1,22 @@
+import { OnInit, Component, Input } from '@angular/core';
+
+@Component({
+    selector: 'nav-item',
```
IMHO should be app-nav-item

cgolian commented on src/app/dashboard/components/dashboard/dashboard.component.ts:
```diff
@@ -0,0 +1,34 @@
+import { BreakpointObserver } from '@angular/cdk/layout';
+
+@Component({
+    selector: 'dashboard',
```
IMHO should be app-dashboard

cgolian commented on src/app/dashboard/components/appbar/appbar.component.ts:
```diff
@@ -0,0 +1,31 @@
+import { MatSidenav } from '@angular/material/sidenav';
+
+@Component({
+    selector: 'appbar',
```
IMHO should be app-appbar

cgolian commented on src/app/dashboard/components/dashboard/dashboard.component.html:
```diff
@@ -0,0 +1,24 @@
+<div>
```
We should decide whether we are going to be using inline templates or separate template files and then go with one approach

cgolian commented on src/app/dashboard/components/dashboard/dashboard.component.ts:
```diff
@@ -0,0 +1,34 @@
+    sidenavMode: string;
+    sidenavOpened: boolean;
+
+    constructor(public breakpointObserver: BreakpointObserver) {
```
does breakpointObserver has to be public?

mlukac-senacor committed:
SCNG-2
Add app prefix to selectors.
Use template instead of separate html file.
Use private modified in constructors.

mlukac-senacor committed:
SCNG-2
Add layout module.
Fix routing.

cgolian commented on src/app/login/login.module.ts:
```diff
@@ -23,16 +23,17 @@ import {LoginFormComponent} from './components/login-form/login-form.component';
     MatIconModule,
-    TranslateModule
+    TranslateModule,
+    RouterModule.forChild([{ path: '', component: LoginComponent }])
```
this line of code is correct but I have never seen routes defined in this way. It should be in a separate "routes" variable IMO (or even better in a router module, because it is done that way elsewhere - LayoutRoutingModule).

cgolian commented on src/app/layout/layout.module.ts:
```diff
@@ -0,0 +1,36 @@
+    LayoutRoutingModule
+  ],
+  exports: [
+    LayoutComponent
```
why is LayoutComponent exported here? as far as I can see it is only used inside of LayoutModule so it it not necessary.

cgolian commented on src/app/dashboard/dashboard.module.ts:
```diff
@@ -0,0 +1,21 @@
+  imports: [
+    CommonModule,
+    TranslateModule,
+    RouterModule.forChild([{ path: '', component: DashboardComponent }])
```
What I don't like is that routes are not declared in one way. If we have already used a routing module (LayoutRoutingModule) we should do it in the same way also here.

cgolian commented on src/app/dashboard/dashboard.module.ts:
```diff
@@ -0,0 +1,21 @@
+    RouterModule.forChild([{ path: '', component: DashboardComponent }])
+  ],
+  exports: [
+    DashboardComponent,
```
why is DashboardComponent exported here? it is only used in DashboardModule

cgolian reviewed (commented):
found a couple of issues concerning exports and how routes are declared, otherwise it's fine 👍

mlukac-senacor committed:
SCNG-2
Add separate routing modules.
Remove exports if not necessary.
